contracts/DefaultStrategy.sol

```diff
@@ -522,248 +436,288 @@ contract DefaultStrategy is UUPSOwnableUpgradeable, UsingRegistryUpgradeable, Ma
-     * @return deprecatedGroupsWithdrawn The array of deprecated groups to be
-     * withdrawn from.
-     * @return deprecatedWithdrawalsPerGroup The amount of votes to withdraw
-     * from the respective deprecated group in `deprecatedGroupsWithdrawn`.
-     * @return numberDeprecatedGroupsWithdrawn The number of groups in
-     * `deprecatedGroupsWithdrawn` that have a non zero withdrawal.
-     * @return remainingWithdrawal The number of votes that still need to be
-     * withdrawn after withdrawing from deprecated groups.
-     * @dev Non zero entries of `deprecatedWithdrawalsPerGroup` will be exactly
-     * a prefix of length `numberDeprecatedGroupsWithdrawn`.
-     */
-    function getDeprecatedGroupsWithdrawalDistribution(uint256 withdrawal)
-        internal
-        returns (
-            address[] memory deprecatedGroupsWithdrawn,
-            uint256[] memory deprecatedWithdrawalsPerGroup,
-            uint256 numberDeprecatedGroupsWithdrawn,
-            uint256 remainingWithdrawal
-        )
+     * @notice Returns expected stCELO vs actual stCELO for group.
+     * @param group The group.
+     * @return expectedStCelo The stCELO which group should have.
+     * @return actualStCelo The stCELO which group has.
+     */
+    function getExpectedAndActualStCeloForGroup(address group)
+        public
+        view
+        returns (uint256 expectedStCelo, uint256 actualStCelo)
     {
-        remainingWithdrawal = withdrawal;
-        uint256 numberDeprecatedGroups = deprecatedGroups.length();
-        deprecatedGroupsWithdrawn = new address[](numberDeprecatedGroups);
-        deprecatedWithdrawalsPerGroup = new uint256[](numberDeprecatedGroups);
-        numberDeprecatedGroupsWithdrawn = 0;
-
-        for (uint256 i = 0; i < numberDeprecatedGroups; i++) {
-            numberDeprecatedGroupsWithdrawn++;
-            deprecatedGroupsWithdrawn[i] = deprecatedGroups.at(i);
-            uint256 currentStCeloInStrategy = defaultStrategyTotalStCeloVotes[
-                deprecatedGroupsWithdrawn[i]
-            ];
-            uint256 currentVotes = IManager(manager).toCelo(currentStCeloInStrategy);
-            deprecatedWithdrawalsPerGroup[i] = Math.min(remainingWithdrawal, currentVotes);
-            remainingWithdrawal -= deprecatedWithdrawalsPerGroup[i];
-
-            if (currentVotes == deprecatedWithdrawalsPerGroup[i]) {
-                if (!deprecatedGroups.remove(deprecatedGroupsWithdrawn[i])) {
-                    revert FailedToRemoveDeprecatedGroup(deprecatedGroupsWithdrawn[i]);
-                }
-                emit GroupRemoved(deprecatedGroupsWithdrawn[i]);
-            }
+        address head = activeGroups.getHead();
+        expectedStCelo = totalStCeloInDefaultStrategy / activeGroups.getNumElements();
+        if (group == head) {
+            uint256 divisionResidue = totalStCeloInDefaultStrategy -
+                (expectedStCelo * activeGroups.getNumElements());
```

Comment:
Potential gas saving: assign `activeGroups.getNumElements()` to a memory variable to avoid an extra storage read (assuming compiler doesn't optimize this).